File: li/cil/oc/api/driver/Slot.scala
package li.cil.oc.api.driver

/**
 * List of possible item component types.
 * <p/>
 * This is used to determine which item components may go into which slots in
 * a computer, since unlike block components, item components must be placed
 * inside the computer, not next to it.
 */
object Slot extends Enumeration {
  /** Power generating components, such as generators. */
  val Power = Value("Power")

  /** Memory extension components. */
  val Memory = Value("Memory")

  /** Hard disk drives. */
  val HardDiskDrive = Value("HardDiskDrive")

  /** Extension cards such as graphics or redstone cards. */
  val Card = Value("Card")

  /** Floppy disks. */
  val Disk = Value("Disk")
}

File: li/cil/oc/api/driver/Slot.java
package li.cil.oc.api.driver;

/**
 * List of possible item component types.
 * <p/>
 * This is used to determine which item components may go into which slots in
 * a computer's and robot's inventory.
 */
public enum Slot {
    /**
     * Invalid slot type.
     */
    None,

    /**
     * Extension cards such as graphics cards or redstone cards.
     */
    Card,

    /**
     * Floppy disks. These can be inserted into the Disk Drive block.
     */
    Disk,

    /**
     * Hard disk drives. These can be installed in computers.
     */
    HardDiskDrive,

    /**
     * Memory extension components. Used to increase computer RAM.
     */
    Memory,

    /**
     * Tool slot in robots (equipment slot).
     */
    Tool,

    /**
     * Upgrade slot for robots. For special robot upgrades such as internal
     * engines and the like.
     */
    Upgrade
}